search_service/proxy/atlas.py

```diff
@@ -0,0 +1,230 @@
+        :param field_name: field name to do the searching(e.g schema_name, tag_names)
+        :param field_value: value for the field for filtering
+        :param page_index: index of search page user is currently on
+        :return: SearchResult Object
+        :return:
+        """
+
+        sql = f"Table from Table where false"
+        count_sql = f"{sql} select count()"
+        if field_name == 'tag':
+            sql = f"from Table where Table is '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'schema':
+            sql = f"from Table where db.name like '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'table':
+            sql = f"from Table where name like '{field_value}'"
+            count_sql = f"{sql} select count()"
+        elif field_name == 'column':
+            sql = f"hive_column where name like '{field_value}' select table"
+            # TODO nanne: count tables instead of columns
+            count_sql = f"hive_column where name like '{field_value}' select count()"
+
+        LOGGING.debug(f"Used following sql query: {sql}")
+        tables: List[Table] = []
+        count_value = 0
+        try:
+            # count results
+            count_params = {'query': count_sql}
+            count_results = list(self.atlas.search_dsl(**count_params))[0]
+            count_value = count_results._data['attributes']['values'][0][0]
+
+            params = {'query': f"{sql} limit {self.page_size} offset {page_index * self.page_size}"}
+            search_results = self.atlas.search_dsl(**params)
+            if count_value > 0 and page_index * self.page_size <= count_value:
+                # unpack all collections (usually just one collection though)
+                for collection in search_results:
+                    if hasattr(collection, 'entities'):
+                        tables.extend(self._parse_results(response=collection.entities))
+        except BadRequest:
+            pass
+
+        return SearchResult(total_results=count_value, results=tables)
+
+    @timer_with_counter
+    def fetch_search_results(self, *,
+                             query_term: str,
+                             page_index: int = 0) -> SearchResult:
+        """
+        Query Atlas and return results as list of Table objects
+        We use the Atlas DSL for querying the tables.
+        https://atlas.apache.org/Search-Advanced.html
+
+        :param query_term: search query term
+        :param page_index: index of search page user is currently on
+        :return: SearchResult Object
+        """
+
+        if not query_term:
+            # return empty result for blank query term
+            return SearchResult(total_results=0, results=[])
+
+        # define query
+        sql = f"Table from Table " \
+            f"where name like '*{query_term}*' or " \
+            f"description like '*{query_term}*' "
+
+        # count amount of tables
```

Comment:
So apparently we do need to run two queries here, one to calculate total results and one to make the page_size results. This is weird. Ideally the search API should return the total count within the page_size results. Can we check if we can implement this in Atlas itself if it's not there already?

Reply:
😢 it is not there unfortunately